api/app/src/routes/user.ts

```diff
@@ -115,6 +117,51 @@ router.get(
   })
 );
 
+/** ---------------------------------------------------------------------------------------
+ * DELETE /user/revoke
+ *
+ * Revoke access to a provider
+ *
+ * @param   {string}  req.query.provider    The provider to revoke access.
+ * @param   {string}  req.query.userId      The internal user ID.
+
+ * @return  {{success: boolean}}      If successfully removed.
+ */
+router.delete(
+  "/revoke",
+  asyncHandler(async (req: Request, res: Response) => {
+    if (!req.query.userId) {
+      return res.sendStatus(status.BAD_REQUEST);
+    }
```

Comment:
Already done on `getUserIdOrFail`

We might want to rename that function to `getUserIdFromQueryOrFail` at some point :)